PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,111 @@
+# Description
+> Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context and any links to documentation created.
+> Reference: [PR Submission Guide](/submission_process.md)
+
+## Type of change
+
+Please delete options that are not relevant.
+
+- [ ] Bug fix (non-breaking change which fixes an issue)
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+- [ ] This change requires a documentation update
+
+### Dependencies
+> List all libraries and imports utilized, e.g. Ownable.sol from OZ
+
+### Externalities
+> List all external (to this PR) contracts that are interacted with - within Yield ecosystem and without.
+
+> _Note to Reviewer: Examine every component the feature may touch. Then go back through excluded components and ensure they cannot be affected._
+
+### Risk 
+> Highlight areas of risk. E.g. Assembly code blocks, use of unchecked, function call made to an external contract that lies outside of the Yield ecosystem. 
+
+> Example: fn A in contractB.sol makes a call to someexternalcontract.sol
+
+## Contracts
+> This section serves as a brief for the reviewer, code author is to introduce contracts and elaborate on salient points.
+
+> ContractA.sol -> description + points
+
+> ContractB.sol -> description + points
+
+> Reference: See section '[New contracts and areas of risk](https://github.com/yieldprotocol/yieldspace-tv/pull/3)'
+
+### Complexity
+> Highlight areas of abstraction/complexity; e.g. functions that have complicated math or are deeply layered.
+
+> Example: ContractA.sol, Function doSomeComplexMath()
+
+## Testing
+> Elaborate on your testing approach - assumptions made, concerns and any caveats.
+
+> Map your state inheritance testing flow. This could be in the form of a [text tree](https://github.com/calnix/Collateralized-Vault/blob/AddedCode/Readme.md#testing) or an illustrated diagram (excalidraw/draw.io).
```

Comment:
this is all great info, but not sure it belongs in the pr template.  also idt it's particularly important to elaborate on your testing approach in many situations.  I am intersted in:

1) If it is a bug, was a test written so that the bug won't happen again??
2) What kind of tests were used?  local, unit, mainnet fork, fuzzing, symbolic execution

Pasting the state inheritance testing form sound pretty cool but also sounds like a bit of a PITA for all but very large projects.  This kind of thing would be nice to have in the testing documentation.